pkg/pool-utils/contracts/test/MockVault.sol

```diff
@@ -224,8 +224,22 @@ contract MockVault is IPoolSwapStructs {
     }
 
     // This supports calls from the VaultReentrancyLib in unit tests, so that they don't revert.
-    // Note that it isn't itself protected by the Vault context, like the real Vault would be.
-    function manageUserBalance(IVault.UserBalanceOp[] memory ops) external payable {
+    function manageUserBalance(IVault.UserBalanceOp[] memory) external payable nonReentrant {
         // solhint-disable-previous-line no-empty-blocks
     }
+
+    // The real Vault doesn't have any hooks that are view functions, so add one to this Vault
+    // specifically to test read-only reentrancy protection on view functions.
+    function functionWithHook(address pool) external nonReentrant {
+        (bool success, bytes memory returnData) = pool.call(abi.encodeWithSignature("viewHook()"));
```

Comment:
We could also call `protectedViewFunction` directly right? Or am I missing something?

Reply:
I'm just making it match the pattern of the others (i.e., be more realistic).